culturefoundry/static_el_controller
# Cf customizations

## TRELLO CARD
https://trello.com/c/L2jAaF0A/3488-cf-create-a-way-for-elements-templates-chunks-snippets-and-plugins-to-opt-in-to-be-edited-exclusively-via-version-controlled-cap

## DESCRIPTION
As you can see, this is a fork of the unmaintained extra, StaticSaver. I changed the name to StaticElController to reflect what it does more accurately and to indicate its future direction as separate from the original. This effectively uses the original efforts of StaticSaver as a base, but builds on and alters its functionality to meet our needs. 

This version adds new system settings, adds JS to lock editing of static files, and writes the README (which should give you a good overview of what the Extra actually accomplishes).

## TESTING
I've set up the new MODx version control workflow on the Culture Foundry website. This extra is installed on Dev and Prod, and all elements have been dumped as a base. To test:

- Create a new chunk in production with some basic html content. 
- Add the chunk somewhere in this "playground" `StaticElController` template I added (`/manager/?a=element/template/update&id=25`)
- You should see the chunk content appearing where you added it on this page: `/staticelcontroller-test?preview=true` (I set up a hidden from menus resource to use the `StaticElController` template)
- Mark the chunk as static, note the path and media source that are added automatically (it should match the paths of the corresponding/existing elements from dumping chunks)
- Repeat the previous 4 steps on dev

- `cfdb culturefoundry --fresh`
- `git pull origin master`
- Edit the new chunk you created earlier locally in your text editor
- Commit and push to master -- you should see the new chunk content appearing on the playground page on dev
- `bundle exec cap deploy` -- you should now see the new chunk content appearing on the playground page on prod
